From 788fba6b68128c4667697d5e83981e7aa8c99f2d Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:50:01 +0800
Subject: [PATCH 48/62] Revert "drm/komeda: Fix an undefined behavior bug in
 komeda_plane_add()"

This reverts commit 78a3e9fcdb7b1712de30d9d248bf19cee6a890b0.
---
 drivers/gpu/drm/arm/display/komeda/komeda_plane.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
index bc3f42e915e9..a5f57b38d193 100644
--- a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
+++ b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
@@ -264,10 +264,6 @@ static int komeda_plane_add(struct komeda_kms_dev *kms,
 
 	formats = komeda_get_layer_fourcc_list(&mdev->fmt_tbl,
 					       layer->layer_type, &n_formats);
-	if (!formats) {
-		kfree(kplane);
-		return -ENOMEM;
-	}
 
 	err = drm_universal_plane_init(&kms->base, plane,
 			get_possible_crtcs(kms, c->pipeline),
-- 
2.27.0

